File: Automation/001-AutomationFramework/C#/AutomationFramework/Engines/SeleniumEngine/DriverInitialization/Browsers/Chrome.cs
using CommonCore.Tests;
using OpenQA.Selenium;
using OpenQA.Selenium.Chrome;
using Utilities;
using WebDriverManager.DriverConfigs.Impl;

namespace SeleniumEngine.DriverInitialization.Browsers
{
    public class Chrome : IBrowserDriver
    {
        public virtual IWebDriver Initialize()
        {
            ChromeOptions chromeOptions = new ChromeOptions();

            if (!TestInMemoryParameters.Instance.MultipleBrowserInstances)
            {
                WindowsProcessManager.KillProcess(TestInMemoryParameters.Instance.DriverType, "chrome");
            }

            chromeOptions.SetLoggingPreference(LogType.Browser, LogLevel.All);
            chromeOptions.AddUserProfilePreference("download.prompt_for_download", true);
            chromeOptions.AddUserProfilePreference("download.default_directory", Directory.GetCurrentDirectory());
            chromeOptions.AddUserProfilePreference("profile.default_content_setting_values.automatic_downloads", 1);

            chromeOptions.AddArguments("--disable-popup-blocking",
                                       "--disable-extensions",
                                       "--start-maximized",
                                       "--safebrowsing-disable-download-protection",
                                       "--no-sandbox",
                                       "--disable-dev-shm-usage",
                                       "--whitelisted-ips",
                                       "--ignore-ssl-errors=yes",
                                       "--ignore-certificate-errors",
                                       "--disable-gpu");
            new WebDriverManager.DriverManager().SetUpDriver(new ChromeConfig());
            return new ChromeDriver(chromeOptions);
        }
    }
}

File: Automation/TestPapas/TestAutomationFramework/src/main/java/engines/selenium/driverInitialization/browsers/Chrome.java
package engines.selenium.driverInitialization.browsers;

import engines.selenium.driverInitialization.BrowserDriver;
import engines.selenium.driverInitialization.WebDriverFactory;
import org.openqa.selenium.PageLoadStrategy;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.chrome.ChromeOptions;
import org.openqa.selenium.remote.RemoteWebDriver;

import java.nio.file.Path;

public class Chrome implements BrowserDriver {
    public static RemoteWebDriver initialize() {
        ChromeOptions chromeOptions = new ChromeOptions();
        String driverLocation = Path.of("").toAbsolutePath() + "/src/main/java/engines/selenium/driverInitialization/browserDrivers/chrome/chromedriver.exe";
        System.setProperty("webdriver.chrome.driver" ,  driverLocation);

        chromeOptions.addArguments("--disable-popup-blocking",
                "--disable-extensions",
                "--start-maximized",
                "--disable-blink-features=AutomationControlled",
                "--safebrowsing-disable-download-protection",
                "--no-sandbox",
                "--disable-dev-shm-usage",
                "--whitelisted-ips",
                "--ignore-ssl-errors=yes",
                "--ignore-certificate-errors",
                "--disable-gpu");
        chromeOptions.setExperimentalOption("profile.default_content_setting_values.media_stream_mic", 1);
        chromeOptions.setExperimentalOption("profile.default_content_setting_values.media_stream_camera", 1);
        chromeOptions.setExperimentalOption("profile.default_content_setting_values.geolocation", 0);
        chromeOptions.setExperimentalOption("profile.default_content_setting_values.notifications", 1);

        WebDriverFactory.getThreadSafeInstance().driver = new ChromeDriver(chromeOptions);

        return WebDriverFactory.getThreadSafeInstance().driver;
    }
}
